pandas/tests/io/parser/test_network.py

```diff
@@ -50,151 +93,142 @@ def check_compressed_urls(salaries_table, compression, extension, mode,
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df_reader = read_csv('s3://pandas-test/tips.csv' + ext,
-                                 chunksize=chunksize, compression=comp,
-                                 engine='python')
-            assert df_reader.chunksize == chunksize
-            for i_chunk in [0, 1, 2]:
-                # Read a couple of chunks and make sure we see them properly.
-                df = df_reader.get_chunk()
-                assert isinstance(df, DataFrame)
-                assert not df.empty
-                true_df = local_tips.iloc[
-                    chunksize * i_chunk: chunksize * (i_chunk + 1)]
-                tm.assert_frame_equal(true_df, df)
-
-    @tm.network
-    def test_parse_public_s3_bucket_python(self):
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df = read_csv('s3://pandas-test/tips.csv' + ext, engine='python',
-                          compression=comp)
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')), df)
-
-    @tm.network
-    def test_infer_s3_compression(self):
-        for ext in ['', '.gz', '.bz2']:
-            df = read_csv('s3://pandas-test/tips.csv' + ext,
-                          engine='python', compression='infer')
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')), df)
-
-    @tm.network
-    def test_parse_public_s3_bucket_nrows_python(self):
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df = read_csv('s3://pandas-test/tips.csv' + ext, engine='python',
-                          nrows=10, compression=comp)
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')).iloc[:10], df)
 
-    @tm.network
-    def test_s3_fails(self):
-        with pytest.raises(IOError):
-            read_csv('s3://nyqpug/asdf.csv')
+def test_s3_fails(test_s3_resource):
+    with pytest.raises(IOError):
+        read_csv('s3://nyqpug/asdf.csv')
 
-        # Receive a permission error when trying to read a private bucket.
-        # It's irrelevant here that this isn't actually a table.
-        with pytest.raises(IOError):
-            read_csv('s3://cant_get_it/')
+    # Receive a permission error when trying to read a private bucket.
+    # It's irrelevant here that this isn't actually a table.
+    with pytest.raises(IOError):
+        read_csv('s3://cant_get_it/')
 
-    @tm.network
-    def boto3_client_s3(self):
-        # see gh-16135
 
-        # boto3 is a dependency of s3fs
-        import boto3
-        client = boto3.client("s3")
+def test_read_csv__handles_boto_s3_object(test_s3_resource, tips_file):
```

Comment:
extra `_` between `csv` and `handles`

Reply:
That's a habit I've formed at my current position. We name our tests like "test_name_of_function__the_thing_your_testing". I can remove that.